pkgs/tools/misc/figurine/default.nix

```diff
@@ -0,0 +1,29 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildGoModule
+}:
+
+let
+  version = "1.3.0";
+in
+buildGoModule {
+  pname = "figurine";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "arsham";
+    repo = "figurine";
+    rev = "v${version}";
+    hash = "sha256-1q6Y7oEntd823nWosMcKXi6c3iWsBTxPnSH4tR6+XYs=";
+  };
+
+  vendorSha256 = "sha256-mLdAaYkQH2RHcZft27rDW1AoFCWKiUZhh2F0DpqZELw=";
+
+  meta = with lib; {
```

Comment:
```suggestion
  ldflags = [
    "-s"
    "-w"
    "-X main.version=${version}"
    "-X main.currentSha=${src.rev}"
  ];

  meta = with lib; {
```

Those `ldflags` are used in the upstream https://github.com/arsham/figurine/blob/v1.3.0/Makefile#L61 and provide version info for the `version` sub-command.


```bash
# before
❯ result/bin/figurine version
figurine version development (N/A)

# after
❯ result/bin/figurine version
figurine version 1.3.0 (v1.3.0)
```

While upstream does technically specify different values using `git describe` and `git rev-parse`, which we could do as well, this is much easier and is usually good enough.